lib/semantics/resolve-names.cc

```diff
@@ -1334,7 +1330,7 @@ bool ModuleVisitor::Pre(const parser::Submodule &x) {
   }
   PushScope(*parentScope);  // submodule is hosted in parent
   auto &symbol{BeginModule(name, true, subpPart)};
-  if (!ancestor->AddSubmodule(name, &CurrScope())) {
+  if (!ancestor->AddSubmodule(name, &currScope())) {
```

Comment:
Maybe AddSubmodule() should take a reference argument rather than a pointer, since it can't be null.